Sources/SwiftFormat/PrettyPrint/TokenStreamCreator.swift

```diff
@@ -68,17 +69,32 @@ fileprivate final class TokenStreamCreator: SyntaxVisitor {
   /// in a function call containing multiple trailing closures).
   private var forcedBreakingClosures = Set<SyntaxIdentifier>()
 
-  init(configuration: Configuration, operatorTable: OperatorTable) {
+  /// Tracks whether we last considered ourselves inside the selection
+  private var isInsideSelection = true
+
+  init(configuration: Configuration, selection: Selection?, operatorTable: OperatorTable) {
     self.config = configuration
+    self.selection = selection
     self.operatorTable = operatorTable
     self.maxlinelength = config.lineLength
     super.init(viewMode: .all)
   }
 
   func makeStream(from node: Syntax) -> [Token] {
+    // if we have a selection, then we start outside of it
+    if selection != nil {
+      appendToken(.disableFormatting(AbsolutePosition(utf8Offset: 0)))
+      isInsideSelection = false
+    }
+
     // Because `walk` takes an `inout` argument, and we're a class, we have to do the following
     // dance to pass ourselves in.
     self.walk(node)
+
+    // Make sure we output any trailing text after the last selection range
+    if selection != nil {
+      appendToken(.enableFormatting(nil))
```

Comment:
Do we also need to reset `isInsideSelection` to `true` here?

Also: Do you know if `makeStream` ever gets called for more than one node so that this `.enableFormatting` actually matters?